File: LeetCode/Medium/0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        if(nums.length==1) return nums[0];
        int l=0;
        int r=nums.length-1;
        while(l<r){
            int mid=l+(r-l)/2;
            if(nums[mid]==nums[mid+1]){
                if(mid%2==0){
                    l=mid+2;
                }
                else{
                    r=mid-1;
                }
            }
            else if(nums[mid]==nums[mid-1]){
                if(mid%2==0){
                    r=mid;
                }
                else{
                    l=mid+1;
                }
            }
            else{
                return nums[mid];
            }
        }
        return nums[l];
    }
}

File: LeetCode/Medium/0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.cs
public class Solution {
    public int SingleNonDuplicate(int[] nums) {
        if(nums.Length==1) return nums[0];
        int l=0;
        int r=nums.Length-1;
        while(l<r){
            int mid=l+(r-l)/2;
            if(nums[mid+1]==nums[mid]){
                mid++;
            }
            else if(nums[mid-1]!=nums[mid]){
                return nums[mid];
            }
            if(mid%2==0){
                r=mid-2;
            }
            else{
                l=mid+1;
            }

        }
        return nums[l];
    }
}